docs/integrations/xgboost.rst

```diff
@@ -0,0 +1,189 @@
+Automatic logging of the XGBoost training
+=========================================
+.. image:: ../_static/images/xgboost/xgboost_0.png
+   :target: ../_static/images/xgboost/xgboost_0.png
+   :alt: XGBoost overview
+
+XGBoost is an optimized distributed gradient boosting library designed to be highly efficient, flexible and portable. Integration with |Neptune| lets you log multiple training artifacts out-of-the-box. It is implemented as XGBoost callback and includes:
+
+* |metrics| (train and eval) after each boosting iteration.
+* |model| (Booster) to Neptune after last boosting iteration.
+* |feature| to Neptune as image after last boosting iteration.
+* |tree| to Neptune as images after last boosting iteration.
+
+.. tip:: Try integration right away with this |google-colab|.
+
+Prerequisites
+-------------
+This integration makes use of the XGBoost library and is part of the |neptune-contrib|.
+
+Make sure you have all dependencies installed. You can use bash command below:
+
+.. code-block:: bash
+
+    pip install neptune-client pandas xgboost scikit-learn matplotlib graphviz neptune-contrib[monitoring]
+
+Basic example
+-------------
+Make sure you created an experiment before you start XGBoost training. Use |create-experiment|.
+
+Here is how to use Neptune-XGBoost integration:
+
+.. code-block:: python3
+
+    import neptune
+    ...
+    # here you import `neptune_callback` that does the magic (the open source magic :)
+    from neptunecontrib.monitoring.xgboost_monitor import neptune_callback
+
+    ...
+
+    # Use neptune callback
+    neptune.create_experiment(name='xgb', tags=['train'], params=params)
+    xgb.train(params, dtrain, num_round, watchlist,
+              callbacks=[neptune_callback()])  # neptune_callback is here
```

Comment:
I am wondering whether it should be `neptune_callback` or `NeptuneCallback` let's leave it as is but we should raise an issue and think about standardization for the names/conventions here.

Reply:
I follow PEP 8 guideline. For methods it is https://www.python.org/dev/peps/pep-0008/#function-and-variable-names, so `neptune_callback` is correct name.

If it's a class then NeptuneCallback is correct. Here, it's not a class.